### ISSUE TITLE
Can crash the debugger after running this program many times with Ctrl+Shift+F5

### ISSUE THREAD

**Issue by mistymntncop** [EpicGamesExt/raddebugger#140]
Tested this on the latest master and dev branches.

Steps to reproduce:

1. Compile this program with MSVC - normal flags e.g. (cl -Zi -Od test.c)  

```
#include <stdio.h>
#include <stdint.h>
#include <windows.h>

DWORD WINAPI thread_func(VOID *param) {
    while(TRUE) {
        int lol = 3;
    }
}

int main(void) {
    DWORD thread_id = 0;
    HANDLE thread_handle = CreateThread(0, 0, thread_func, 0, 0, &thread_id);
    
    return 0;
}

```
3. Put a breakpoint on line "int lol = 3;"
4. Hold down the Restart debugging key (Ctrl+Shift+F5) for 30 seconds (or longer). Eventually you will get the following error message:

```
[Window Title]
Fatal Exception

[Content]
A fatal exception (code 0xc0000005) occurred. The process is terminating.

Press Ctrl+C to copy this text to clipboard, then create a new issue in
https://github.com/EpicGames/raddebugger/issues

Call stack:
0. [0x7ff71df054db] demon_run +507, demon_core.c line 289
1. [0x7ff71df2c0ef] ctrl_thread__next_demon_event +2991, ctrl_core.c line 1730
5. [0x7ff71df2d52d] ctrl_thread__launch_and_init +461, ctrl_core.c line 2069
6. [0x7ff71df2b3c2] ctrl_thread__entry_point +418, ctrl_core.c line 1544
7. [0x7ff71de977f8] w32_thread_base +56, os_core_win32.c line 173
8. [0x7ffa28077344] KERNEL32
9. [0x7ffa29b226b1] ntdll
```

edit: looks like this in remedy when I attach to the crashed process:
![image](https://github.com/EpicGames/raddebugger/assets/105329747/a27268b2-3f6e-429b-9565-ae1f0fb8da18)

[This line](https://github.com/EpicGames/raddebugger/blob/f9e2aa99a9bd604463be48bff950bc1f46fd427f/src/ctrl/ctrl_core.c#L2023) decrements run_ctrls.run_entity_count when it is zero. For some reason the "ctrl_thread__launch_and_init" function is encountering a DEMON_EventKind_ExitProcess event after all run entities have been removed.

**Comment by mistymntncop** [comment]
I think this crash is just one symptom of a greater issue. Holding (Ctrl+Shift+F5) to restart the debugged process in quick succession eventually somehow leads to the current instance of process not being restarted but new instances being created. See the Scheduler panel here and also the crazy breakpoint hit graphic.

![image](https://github.com/EpicGames/raddebugger/assets/105329747/9b54844b-9639-4368-a591-53d1e2905acc)

**Comment by ryanfleury** [comment]
This has been more thoroughly addressed with 62659c5161ce25a671b6585726f2cbaa531f4f54.
